shaders/EditorHexGrid.shader

```diff
@@ -1,27 +1,53 @@
 shader_type spatial;
 render_mode unshaded;
 
-// TODO: implement our own version of procedural hex grid tiling for a better
+// Implemented our own version of procedural hex grid tiling for a better
 // looking and more uniform hexes.
 // See andrewhungblog.wordpress.com/2018/07/28/shader-art-tutorial-hexagonal-grids/
 
-uniform sampler2D gridTexture;
 uniform sampler2D maskTexture;
 uniform vec4 color : hint_color;
+uniform float lineWidth = 0.02;
+uniform float edgeLength = 1.3;
 
-uniform vec2 gridSize;
+const vec2 hexSize = vec2(1.7320508, 1.0); // 1.7320508 = sqrt(3)
 
 varying vec3 worldPos;
+varying float minLineWidth;
+
+// Returns normalized distance to the nearest hex center.
+float calcHexCenterDistance(vec2 coord)
+{
+    vec4 hexCenter = round(vec4(coord, coord - vec2(1.0, 0.5)) / hexSize.xyxy);
+    vec4 offset = vec4(coord - hexCenter.xy * hexSize, coord - (hexCenter.zw + 0.5) * hexSize);
+    vec2 position = dot(offset.xy, offset.xy) < dot(offset.zw, offset.zw) ? abs(offset.xy) : abs(offset.zw);
+    return max(dot(position, hexSize * 0.5), position.y) + 0.5;
+}
```

Comment:
Please make it more clear that this implementation is an adapted variant of Andrew Hung's code

Reply:
Is there any licensing info available regarding that tutorial? I think it's a bit iffy if we have copypasted an entire function from there without any kind of info regarding what's the license for that code...

Reply:
No mentioning of any licensing on the blog anywhere, but the closer
```
Armed with these functions, you now have everything you need to create some cool and unique hex-based effects.
[...]

Happy shader-writing! The only limit is your imagination. And your coding ability.
```
stongly suggests that he intends that the code he showed is meant to be used in projects

Reply:
Further adopted the code. But still, this is a tutorial. I learned this technique from there. There will always be some recognizable matches. The function is not the same. It is heavily adopted to our use case.

I have absolutely no experience with licenses. So I'm not sure about this.